server/member/member.go

```diff
@@ -115,21 +112,53 @@ func (m *Member) GetLeader() *pdpb.Member {
 	return member
 }
 
-// EnableLeader sets the member to PD leader.
+// setLeader sets the member's PD leader.
+func (m *Member) setLeader(member *pdpb.Member) {
+	m.leader.Store(member)
+}
+
+// UnsetLeader unsets the member's PD leader.
+func (m *Member) unsetLeader() {
+	m.leader.Store(&pdpb.Member{})
+}
+
+// EnableLeader sets the member itself to PD leader.
 func (m *Member) EnableLeader() {
-	m.leader.Store(m.member)
+	m.setLeader(m.member)
 }
 
 // DisableLeader reset the PD leader value.
 func (m *Member) DisableLeader() {
-	m.leader.Store(&pdpb.Member{})
+	m.unsetLeader()
 }
 
 // GetLeaderPath returns the path of the PD leader.
 func (m *Member) GetLeaderPath() string {
 	return path.Join(m.rootPath, "leader")
 }
 
+// GetLeadership returns the leadership of the PD member.
+func (m *Member) GetLeadership() *election.Leadership {
+	return m.leadership
+}
+
+// CampaignLeadership is used to campaign a PD member's leadership
+// and make it become a PD leader.
+func (m *Member) CampaignLeadership(leaseTimeout int64) error {
+	return m.leadership.Campaign(leaseTimeout, m.MemberValue())
+}
+
+// KeepLeadership is used to keep the PD leader's leadership.
+func (m *Member) KeepLeadership(ctx context.Context) {
+	m.leadership.Keep(ctx)
+}
+
+// CheckLeadership is used to check whether a leadership of
+// PD member is still avalibale.
+func (m *Member) CheckLeadership() bool {
+	return m.leadership.Check()
+}
+
```

Comment:
I think we can hide the `LeaderShip` in these methods here.

Reply:
Done.